hydra-node/src/Hydra/Ledger/Cardano.hs

```diff
@@ -614,18 +438,32 @@ fromLedgerUtxo =
   fn i o =
     Map.singleton (fromLedgerTxIn i) (fromLedgerTxOut o)
 
-toLedgerTxOut :: TxOut CtxUTxO Era -> Ledger.TxOut (ShelleyLedgerEra Era)
-toLedgerTxOut = toShelleyTxOut shelleyBasedEra
+fromCardanoApiUtxo :: UTxO AlonzoEra -> Utxo
+fromCardanoApiUtxo = coerce
 
-fromLedgerTxOut :: Ledger.TxOut (ShelleyLedgerEra Era) -> TxOut ctx Era
-fromLedgerTxOut = fromShelleyTxOut shelleyBasedEra
+utxoPairs :: Utxo' out -> [(TxIn, out)]
+utxoPairs = Map.toList . utxoMap
 
-fromCardanoApiUtxo :: Cardano.Api.UTxO AlonzoEra -> Utxo
-fromCardanoApiUtxo = coerce
+prettyUtxo :: (TxIn, TxOut ctx era) -> Text
+prettyUtxo (k, TxOut _ (txOutValueToValue -> v) _) =
+  T.drop 54 (renderTxIn k) <> " ↦ " <> prettyValue v
 
+utxoFromTx :: CardanoTx -> Utxo
+utxoFromTx (Tx body@(ShelleyTxBody _ ledgerBody _ _ _ _) _) =
+  let txOuts = toList $ Ledger.Alonzo.outputs' ledgerBody
+      txIns =
+        [ Ledger.TxIn (toLedgerTxId $ getTxId body) ix
+        | ix <- [0 .. fromIntegral (length txOuts)]
+        ]
+   in fromLedgerUtxo $ Ledger.UTxO $ Map.fromList $ zip txIns txOuts
+
+-- | Select the minimum (by TxIn) utxo entry from the Utxo map.
 --
--- Witnesses
---
+-- This function is partial.
+utxoMin :: Utxo -> Utxo
+utxoMin = Utxo . uncurry Map.singleton . Map.findMin . utxoMap
```

Comment:
I find `min` or `max` on a may confusing. It's an implementation detail for a function which gives one `singleUtxo` from a given `Utxo`? Not sure if that is a better name, but I don't like `utxoMin` all too much :)

Reply:
I mean, I don't disagree, but that was there long before the PR :sweat_smile: